Add Persian (fa) edition for API Security 2023

This-is-Neo opened the pull request:
Add Persian (fa) edition for API Security 2023

This-is-Neo commented:
#134

PauloASilva commented on editions/2023/fa/0xa1-broken-object-level-authorization.md:
```diff
@@ -0,0 +1,89 @@
+<div dir="rtl" align='right'>
```
This is not required since MkDocs already takes care of it:

![image](https://github.com/user-attachments/assets/6e15656f-6812-462b-9d0d-5135b4893270)

 Code blocks are properly handled:
 
![image](https://github.com/user-attachments/assets/71cdff7d-4383-41d2-a94b-f536fd64ffdb)

This-is-Neo replied:
Hi @PauloASilva 

I'm progressing with the revisions and just wanted to confirm a couple of details to ensure everything aligns with your expectations:

You mentioned that "MkDocs already takes care of it." Could you confirm which elements I can leave as-is, so I avoid unnecessary changes?

Regarding the "code blocks," I assume they’re properly handled, but is there anything specific you’d like adjusted, or can I proceed with the current setup?

Once I have your input, I’ll finalize and commit the changes as soon as possible. Feel free to let me know if there's anything else.

Best regards,

This-is-Neo replied:
Hi @PauloASilva ,

I believe I’ve addressed all the revisions you mentioned, and my latest commit is ready for your review. In my last reply, I had a few questions, which I believe I've now resolved—if so, there's no need for further clarification.

Looking forward to your feedback and any additional input you may have.

Thanks,

PauloASilva reviewed (changes requested):
Hi @This-is-Neo,
The PR needs some changes before becoming ready for merge.
Please kindly review my comments.

Remember that you can [render the site locally using docker][1]. To add the Persian translation to the languages selector you should do the following change:

```
diff --git a/editions/2023/mkdocs.yml b/editions/2023/mkdocs.yml
index e4f73e3..505c4d2 100644
--- a/editions/2023/mkdocs.yml
+++ b/editions/2023/mkdocs.yml
@@ -9,3 +9,5 @@ extra:
       lang: en
     - name: Français
       lang: fr
+    - name: Persian
+      lang: fa

```

Also note that we are no longer providing the `ODT`/`PDF` versions of the document due to the additional effort required to keep them up to date. In the future we may consider providing distributable formats as long as they can be compiled from the source.

Cheers,
Paulo A. Silva

[1]: https://github.com/OWASP/API-Security/blob/8b142239c227e190e27776b6de0ef91c8a963318/CONTRIBUTING.md#rendering-the-web-site-locally

This-is-Neo commented:
> Hi @This-is-Neo, The PR needs some changes before becoming ready for merge. Please kindly review my comments.
> 
> Remember that you can [render the site locally using docker](https://github.com/OWASP/API-Security/blob/8b142239c227e190e27776b6de0ef91c8a963318/CONTRIBUTING.md#rendering-the-web-site-locally). To add the Persian translation to the languages selector you should do the following change:
> 
> ```
> diff --git a/editions/2023/mkdocs.yml b/editions/2023/mkdocs.yml
> index e4f73e3..505c4d2 100644
> --- a/editions/2023/mkdocs.yml
> +++ b/editions/2023/mkdocs.yml
> @@ -9,3 +9,5 @@ extra:
>        lang: en
>      - name: Français
>        lang: fr
> +    - name: Persian
> +      lang: fa
> ```
> 
> Also note that we are no longer providing the `ODT`/`PDF` versions of the document due to the additional effort required to keep them up to date. In the future we may consider providing distributable formats as long as they can be compiled from the source.
> 
> Cheers, Paulo A. Silva

Hi @PauloASilva ,
I wanted to let you know that the updates and revisions you requested in this part have been completed and committed.
Please feel free to review them, and let me know if any further adjustments are needed.

Best regards,

This-is-Neo commented on editions/2023/fa/0x00-toc.md:
Hi @PauloASilva ,
I wanted to let you know that the updates and revisions you requested in this part have been completed and committed.
Please feel free to review them, and let me know if any further adjustments are needed.

Best regards,

This-is-Neo commented on editions/2023/fa/0xa1-broken-object-level-authorization.md:
Hi @PauloASilva ,
Just a quick update—I’m actively working on the revisions you requested. I’ll make sure to commit the changes as soon as possible. If there’s anything else you’d like me to address, feel free to let me know!

Best Regards,

This-is-Neo replied:
Hi @PauloASilva ,
I believe I’ve addressed all the revisions you mentioned, and my latest commit is ready for your review. 

Looking forward to your feedback and any additional input you may have.

Thanks,

PauloASilva commented on editions/2023/fa/0x01-about-owasp.md:
```diff
@@ -0,0 +1,66 @@
+# <div dir="rtl">درباره OWASP </div>
```
I don't see any noticeable difference with/without the `<div>`:

![image](https://github.com/user-attachments/assets/8e13df62-686a-4315-be02-41421ab610f1)

Please carefully review the document removing non-required `<div dir="rtl">` throughout it.

PauloASilva commented on editions/2023/fa/0x01-about-owasp.md:
```diff
@@ -0,0 +1,66 @@
+</div>
+
+<div dir="rtl">
+<ul>
+  <li>استانداردها و ابزارهای امنیت اپلیکیشن.</li>
+  <li>کتاب‌هایی درباره تست امنیت اپلیکیشن‌ها، توسعه ایمن کد و بازبینی امنیت کد.</li>
+  <li>ارائه‌ها و <a href="https://www.youtube.com/user/OWASPGLOBAL">ویدئوها</a>.</li>
+  <li><a href="https://owasp.org/index.php/OWASP_Cheat_Sheet_Series">راهنما و برگه تقلب</a> برای بسیاری از موضوعات رایج.</li>
+  <li>کنترل‌ها و کتابخانه‌های استاندارد در حوزه امنیت.</li>
+  <li><a href="https://owasp.org/index.php/OWASP_Chapter">شعب محلی در سرتاسر جهان</a>.</li>
+  <li>تحقیقات به روز و پیشرو در حوزه امنیت.</li>
+  <li><a href="https://owasp.org/index.php/Category:OWASP_AppSec_Conference">کنفرانس‌های تخصصی</a> در سرتاسر جهان.</li>
+  <li><a href="https://lists.owasp.org/mailman/listinfo">یست‌های پست الکترونیک</a> برای ارسال اخبار.</li>
+</ul>
```
I don't see any special requirement to use Markup instead of Markdown [as in the original file](https://raw.githubusercontent.com/OWASP/API-Security/refs/heads/master/editions/2023/en/0x01-about-owasp.md).

Please carefully review the translation source for issues like this one.

PauloASilva commented on editions/2023/fa/0x10-api-security-risks.md:
```diff
@@ -0,0 +1,37 @@
+در این رویکرد، نوع فناوری مورد استفاده و احتمال وقوع آسیب‌پذیری در رتبه ریسک تاثیر ندارند؛ بعبارت دیگر در این روش رتبه بندی ریسک، راهکار مورد استفاده برای ‌‌‌‌پیاده‌سازی API، با رویکردی مستقل از جزئیات فناوری به ارزیابی ریسک می‌پردازد. هرکدام از عوامل یاد شده می‌تواند در پیداکردن و سواستفاده از یک آسیب‌پذیری به مهاجم کمک بسزایی کند. این رتبه بندی تاثیر واقعی بر کسب و کارها را نشان نداده و این ساز ...
+
+## مراجع
+========
```
This is an [alternate syntax for a level 1 heading](https://www.markdownguide.org/basic-syntax/#alternate-syntax) but the `##` says it is a level 2 heading.

Please check [the original file](https://raw.githubusercontent.com/OWASP/API-Security/refs/heads/master/editions/2023/en/0x10-api-security-risks.md) and change accordingly.

PauloASilva commented on editions/2023/fa/0x10-api-security-risks.md:
```diff
@@ -0,0 +1,37 @@
+
+
+
```
Remove

PauloASilva commented on editions/2023/fa/0x10-api-security-risks.md:
```diff
@@ -0,0 +1,37 @@
+
+
+# RISK ریسک های امنیتی API
+========
```
Not required: this is an [alternate syntax for `#`](https://www.markdownguide.org/basic-syntax/#alternate-syntax).

Please review the translation source for issues likes this one.

PauloASilva commented on editions/2023/fa/0x11-t10.md:
```diff
@@ -0,0 +1,19 @@
+
+
```
Remove

PauloASilva commented on editions/2023/fa/0xa1-broken-object-level-authorization.md:
```diff
@@ -0,0 +1,50 @@
+
```
Remove

PauloASilva commented on editions/2023/fa/0xa1-broken-object-level-authorization.md:
```diff
@@ -0,0 +1,50 @@
+
+# API1:2023  نقض مجوزدهی در سطح اشیاء 
+---
```
No need: check the original (`en`) markdown source

PauloASilva commented on editions/2023/fa/0xa1-broken-object-level-authorization.md:
```diff
@@ -0,0 +1,50 @@
+| دسترسی غیرمجاز می‌تواند منجر به افشای اطلاعات، از دست رفتن داده یا دستکاری آن شود. همچنین دسترسی غیرمجاز به اشیا می‌تواند سبب کنترل کامل حساب کاربری توسط مهاجم گردد. | این حمله رایج‌ترین ‌آسیب‌پذیری APIها بوده و بیشترین پیامدها را نیز در پی دارد. مکانیزم‌های مجوزدهی و کنترل دسترسی در اپلیکیشن‌های مدرن، پیچیده و گسترده هستند. حتی اگر اپلیکیشن زیرساخت مناسب را برای کنترل‌های مجوزدهی ‌‌‌‌پیاده‌ساز ...
+
+### آیا API از نظر نقض مجوزدهی در سطح اشیاء آسیب‌پذیر است؟
+---
```
No need: check the original (`en`) markdown source

PauloASilva commented on editions/2023/fa/0xa1-broken-object-level-authorization.md:
```diff
@@ -0,0 +1,50 @@
+در مسئله‌ی Broken Object Level Authorization (BOLA)، امنیت کاربران در دسترسی به اطلاعات و منابع در سیستم به خطر می‌افتد. این مشکل زمانی رخ می‌دهد که سیستم یک درخواست API حاوی یک شناسه (مثلاً شناسه یک مورد یا اشیاء خاص) را دریافت می‌کند و بدون بررسی دقیق این شناسه و اعتبارسنجی آن، به منابع مرتبط با آن شناسه دسترسی می‌دهد. مهاجمان با تغییر شناسه در درخواست‌های خود می‌توانند به اطلاعاتی دسترسی پیدا  ...
+
+### مثال‌هایی از سناریوهای حمله
+---
```
No need: check the original (`en`) markdown source

PauloASilva commented on editions/2023/fa/0xa1-broken-object-level-authorization.md:
```diff
@@ -0,0 +1,50 @@
+یک سرویس ذخیره‌سازی اسناد آنلاین به کاربران این امکان را می‌دهد که اسناد خود را مشاهده، ویرایش، ذخیره و حذف کنند. هنگامی که کاربری یکی از اسناد خود را حذف می‌کند، یک عملیات درخواستی به نام GraphQL Mutation با استفاده از شناسه (ID) مربوط به سند حذف‌شده به API ارسال می‌شود. این درخواست GraphQL به API اطلاع می‌دهد که یک سند باید حذف شود و API مسئول انجام این عملیات حذف است.
+
+### چگونه از آسیب‌پذیری مجوزدهی نادرست در سطح اشیاء پیشگیری کنیم؟
+---
```
No need: check the original (`en`) markdown source

PauloASilva commented on editions/2023/fa/0xa1-broken-object-level-authorization.md:
```diff
@@ -0,0 +1,50 @@
+- طراحی آزمون‌هایی برای ارزیابی صحت عملکرد مکانیزم‌های مجوزدهی.
+
+### مراجع
+---
```
No need: check the original (`en`) markdown source

PauloASilva commented on editions/2023/fa/0xa1-broken-object-level-authorization.md:
```diff
@@ -0,0 +1,50 @@
+### مراجع
+---
+
+- [Authorization Cheat Sheet](https://cheatsheetseries.owasp.org/cheatsheets/Authorization_Cheat_Sheet.html)
+- [Authorization Testing Automation Cheat Sheet](https://cheatsheetseries.owasp.org/cheatsheets/Authorization_Testing_Automation_Cheat_Sheet.html)
+
+#### خارجی
+- [CWE-285: Improper Authorization](https://cwe.mitre.org/data/definitions/285.html)
+- [CWE-639: Authorization Bypass Through User-Controlled Key](https://cwe.mitre.org/data/definitions/639.html)
+
```
Consider moving the URLs to the bottom of the source file as [in the original EN source](https://raw.githubusercontent.com/OWASP/API-Security/refs/heads/master/editions/2023/en/0xa1-broken-object-level-authorization.md).

If you decide to adhere then review and change the whole translation source accordingly.

PauloASilva reviewed (changes requested):
Hi @This-is-Neo,
Carefully review my comments: some requested changes apply to all translation sources.

Please compare you sources with the original ones in terms of structure e.g. headings, unnecessary `---`, use of markup instead of markdown when not required, ...

Take your time to carefully review the translation source making sure it follows the original `EN` source as much as possible. Keep in mind that reviews like this one require a lot of time and effort from all of us.

Cheers,
Paulo A. Silva

This-is-Neo commented:
> Hi @This-is-Neo, Carefully review my comments: some requested changes apply to all translation sources.
> 
> Please compare you sources with the original ones in terms of structure e.g. headings, unnecessary `---`, use of markup instead of markdown when not required, ...
> 
> Take your time to carefully review the translation source making sure it follows the original `EN` source as much as possible. Keep in mind that reviews like this one require a lot of time and effort from all of us.
> 
> Cheers, Paulo A. Silva



> Hi @This-is-Neo, Carefully review my comments: some requested changes apply to all translation sources.
> 
> Please compare you sources with the original ones in terms of structure e.g. headings, unnecessary `---`, use of markup instead of markdown when not required, ...
> 
> Take your time to carefully review the translation source making sure it follows the original `EN` source as much as possible. Keep in mind that reviews like this one require a lot of time and effort from all of us.
> 
> Cheers, Paulo A. Silva

Hi @PaulaSilva,

I have completed the revisions as requested. Below is a summary of the updates I made:

- Corrected the links for consistency and accuracy.
- Updated the mkdocs.yml file to ensure proper structure and navigation.
- Removed the index.md file as it is no longer required.
- Edited 0x00-header.md to exactly match the original reference in [this link](https://github.com/OWASP/API-Security/blob/master/editions/2023/en/0x00-header.md?plain=1). The original reason I had changed it earlier was due to differences between the Mkdocs server output and the GitHub display, but I have reverted it to the original for consistency.
- Removed all <div dir="rtl"> tags to align with the Markdown structure used in the English version.
- Eliminated all inline markups that were not necessary.
- Corrected the heading levels throughout the document to ensure proper hierarchy and readability.
- Removed any unnecessary empty lines and standardized the use of headings (#) without alternate syntax.
- Moved all URLs to the bottom of the source file, as per the original English source for cleaner and more organized code.

These changes should now make the document fully aligned with the original English source and address all of your previous comments. Please feel free to review and let me know if any further adjustments are needed.

Best regards,

PauloASilva commented:
> These changes should now make the document fully aligned with the original English source and address all of your previous comments. Please feel free to review and let me know if any further adjustments are needed.

Unfortunately, they don't: build process is broken I guess due to [`editions/2023/mkdocs.yml` changes](https://github.com/OWASP/API-Security/pull/134/files#diff-a7154aa043994cee6debcb43bbf902f3764f430292767e84696da70f91c6bcf6).

I may take a while to put this together so that it can be merged.

Cheers,
Paulo A. Silva
